### DISCUSSION TITLE
CORS issue

### DISCUSSION THREAD

**Discussion by JSmithOner** [supabase/supabase#29707] [Questions]
I'm using angular and recently got this message

`Access to fetch at 'https://my-project-id.supabase.co/auth/v1/user' from origin 'http://localhost:4200' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

I've added 

```
  "overrides": {
    "@supabase/supabase-js": {
      "@supabase/auth-js": "2.61.0"
    }
  }
```
in my package.json (as mentionned in [this issue](https://github.com/supabase/supabase-js/issues/936)  ) as I got a bug specific to angular, this is the first time it happens to me
Please let me know if you need more infos

**Comment by javim89** [comment]
Hey! I've got a similar issue.

Did you fixed it?

**Reply by JSmithOner** [reply]
I called the support an the problem disappeared for 1 hour then went back again and I waited approximately 1 day and the rpoblem disappeared completely.I don't know more.But you can always call the support just to tell them you also have the problem.Is it a new project of yours or did it happened on an long time existing project?

**Reply by javim89** [reply]
Ok. thanks. If this week I can't fix I'm going to call the support.

It's an own project. I didn't release to production yet, but yesterday I upgrade the plan to have custom domain and after that I started with the problem.

I don't know if I have a wronk config, or if it's a supabase issue...

**Reply by JSmithOner** [reply]
good luck

**Reply by pac2001man** [reply]
I suspect this is a Supabase issue because it used to work. I get the same problem when calling an endpoint using http from a function (not an edge function).
For info, I don't think you can set CORS parameters in the http request when called from a function.
Has anyone been able to reaolve the issue??

**Reply by Karl-EdwardJeanMehu** [reply]
Were you able to solve this issue? This occurs in a new project. CORS are missing in GET responses.

**Reply by bmaitho** [reply]
I'm getting the issue in my app using Supabase as the backend does anyone have a fix for this ?

**Reply by HeshamAbdalla** [reply]
This has fixed the issue for me 

const corsHeaders = {
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Methods': 'POST, OPTIONS',
  'Access-Control-Allow-Headers': 'Content-Type, Authorization, X-Application-Name, apiKey'
};

**Comment by HeshamAbdalla** [comment]
If you corsHeaders looks like this 

const corsHeaders = {
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Methods': 'POST, OPTIONS',
  'Access-Control-Allow-Headers': 'Content-Type, Authorization'
};
Switched it to 
const corsHeaders = {
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Methods': 'POST, OPTIONS',
  'Access-Control-Allow-Headers': 'Content-Type, Authorization, X-Application-Name, apikey'
};

**Comment by ved-harry** [comment]
If your tests are behind some corperate firewall or VPN that may intercept your request, try quit them first and try again.

**Comment by fsdmrobotics-club** [comment]
this is fixed the problem in my side 
```
export const corsHeaders = {
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Headers': 'authorization, x-client-info, apikey, content-type',
};
```
